### PR TITLE
runtime-rs: hypervisor: Remove debug kernel options

### PR DESCRIPTION
**Pull request by jodh-intel**
Removed the following kernel command line options:

- `earlyprintk=ttyS0`
- `initcall_debug`

Both these options are only useful when debugging a guest kernel failure
which is not a common occurrence.

Further, the `earlyprintk=` option can have a large negative performance
impact (it can increase the VM boot time significantly).

If the user wishes to use either of these options, they can add them to the
`kernel_params=` setting in the Kata configuration file's hypervisor
stanza.

Fixes: #7886.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by fidencio** [comment]
@jodh-intel, this looks good to me, but we need to ensure that users will actually have a way to add this to the command-line when, for instance, using kata-deploy.

In order to do so, I'd like to ask you to also follow what I've done here: https://github.com/kata-containers/kata-containers/pull/7854/files

BTW, does this also need changes on the go runtime?  I do think so.

**Comment by fidencio** [comment]
> BTW, does this also need changes on the go runtime? I do think so.

No, I'm wrong, that's not the case. :-)

**Comment by jodh-intel** [comment]
@fidencio - I put details of how users can re-add those options (if required) in the commits.

**Comment by fidencio** [comment]
> @fidencio - I put details of how users can re-add those options (if required) in the commits.

I saw that, thanks for doing so.
What I'm saying is that if a user deploys kata-containers using kata-deploy they simply cannot go there and edit the files, they should pass it via annotations, and that specific annotation is blocked right now.

Thus, I'm asking you to also add that annotation to the list of allowed annotations, as one for virtio_fs_extra_args (for the go runtime) by https://github.com/kata-containers/kata-containers/pull/7854/files.

And here I'm considering we do have such a list for the runtime-rs, if we don't, then just let me know and ignore the message.

**Review by fidencio — APPROVED** [review]
lgtm, thanks @jodh-intel!

File: src/runtime-rs/crates/runtimes/virt_container/Cargo.toml [hunk 1]
```diff
@@ -20,15 +20,16 @@ serde_derive = "1.0.27"
...
+tokio = { version = "1.8.0" }
 toml = "0.4.2"
 url = "2.1.1"
 async-std = "1.12.0"
-tracing = "0.1.36"
 
 agent = { path = "../../agent" }
 common = { path = "../common" }
-hypervisor = { path = "../../hypervisor" }
+# BUG: FIXME: TESTING:
```
Comment (studychao) [inline src/runtime-rs/crates/runtimes/virt_container/Cargo.toml:30]: Why is this introduced ?

Comment (jodh-intel) [reply src/runtime-rs/crates/runtimes/virt_container/Cargo.toml]: Good catch @studychao - fixed! :smile:

**Pull request merged**